src/lib/config.ts

```diff
@@ -67,5 +67,27 @@ const loadConfig = async () => {
 };
 
 export const persistPeers = async () => {
+  await ensurePeersDirectoryExists();
   await fs.writeFile(peersFilePath, JSON.stringify(config.peers, null, 2));
+};
+
+const ensurePeersDirectoryExists = async () => {
+  try {
+    await fs.access(peersFilePath);
+  } catch(err) {
+    if(err.code === 'ENOENT') {
+      await createPeersDirectory();
+    } else {
+      log.warn(`Could not check for existence of peers subdirectory ${err.code}`);
+    }
+  }
+};
+
+const createPeersDirectory = async () => {
+  try {
+    await fs.mkdir(path.parse(peersFilePath).dir, { recursive: true });
+    log.info('Peers subdirectory created');
+  } catch(err) {
```

Comment:
Should we really just be catching the error here and not throwing anything? If it didn't exist, and we failed to create it, we're going to still be in a bad state and not able to write the peers to disk. It seems like a function called `createPeersDirectory` should throw an error if it is not able to create the directory.

Reply:
I gave that some thought. The reason I didn't want to throw an error at that point is because we are doing work to check the conditions for when the file will be saved. If something goes wrong during the check of those conditions we certainly want to place the errors in the log. But we may not want to discard the possibility that saving the file might still work. If that also fails, then in that case I would throw the error.

Reply:
Okay I'll go with that.